dev/src/bulk-writer.ts

```diff
@@ -73,124 +76,91 @@ const RATE_LIMITER_MULTIPLIER = 1.5;
  */
 const RATE_LIMITER_MULTIPLIER_MILLIS = 5 * 60 * 1000;
 
-/*!
- * Used to represent the state of batch.
- *
- * Writes can only be added while the batch is OPEN. For a batch to be sent,
- * the batch must be READY_TO_SEND. After a batch is sent, it is marked as SENT.
- */
-enum BatchState {
-  OPEN,
-  READY_TO_SEND,
-  SENT,
-}
-
 /**
  * Used to represent a batch on the BatchQueue.
  *
  * @private
  */
-class BulkCommitBatch {
-  /**
-   * The state of the batch.
-   */
-  state = BatchState.OPEN;
-
+class BulkCommitBatch extends WriteBatch {
+  constructor(firestore: Firestore, readonly maxBatchSize: number) {
+    super(firestore);
+  }
   // The set of document reference paths present in the WriteBatch.
   readonly docPaths = new Set<string>();
 
   // An array of pending write operations. Only contains writes that have not
   // been resolved.
   private pendingOps: Array<Deferred<BatchWriteResult>> = [];
 
-  private readonly backoff: ExponentialBackoff;
-
-  constructor(
-    private readonly firestore: Firestore,
-    private writeBatch: WriteBatch,
-    private readonly maxBatchSize: number
-  ) {
-    this.backoff = new ExponentialBackoff();
+  _has(documentRef: firestore.DocumentReference): boolean {
```

Comment:
This is a private class and it doesn't need underscores. I don't mind if you leave them for consistency.

Reply:
Removed.